source/particle/generator/ascii_file.cc

```diff
@@ -0,0 +1,173 @@
+                                   "in which the ASPECT source files were located when ASPECT was "
+                                   "compiled. This interpretation allows, for example, to reference "
+                                   "files located in the 'data/' subdirectory of ASPECT. ");
+                prm.declare_entry ("Data file name", "tracer.dat",
+                                   Patterns::Anything (),
+                                   "The name of the tracer file.");
+                prm.leave_subsection();
+              }
+              prm.leave_subsection();
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+        }
+      }
+
+
+      template <int dim>
+      void
+      AsciiFile<dim>::parse_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Tracers");
+          {
+            prm.enter_subsection("Generator");
+            {
+              prm.enter_subsection("Ascii file");
+              {
+                // Get the path to the data files. If it contains a reference
+                // to $ASPECT_SOURCE_DIR, replace it by what CMake has given us
+                // as a #define
+                data_directory        = prm.get ("Data directory");
+                {
+                  const std::string      subst_text = "$ASPECT_SOURCE_DIR";
+                  std::string::size_type position;
+                  while (position = data_directory.find (subst_text),  position!=std::string::npos)
+                    data_directory.replace (data_directory.begin()+position,
+                                            data_directory.begin()+position+subst_text.size(),
+                                            ASPECT_SOURCE_DIR);
+                }
+
+                data_filename    = prm.get ("Data file name");
+              }
+              prm.leave_subsection();
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      ASPECT_REGISTER_PARTICLE_GENERATOR(AsciiFile,
+                                         "ascii file",
+                                         "Generates a distribution of tracers from coordinates "
+                                         "specified in an Ascii data file. The file format is "
+                                         "identical to the one from other ascii data plugins, "
+                                         "except that it only needs to contain as many columns as "
+                                         "dimensions to specify the position of each tracer. Note "
+                                         "that this plugin does ignore the number of tracers set in "
```

Comment:
I thought about it myself, but I do not quite agree on this. If I am an user, who prepared a tracer file, I would rarely want to create less tracers than specified in my file (then I could have generated a shorter file to begin with). But if I specify a number of tracers that is larger than my file, this is most likely an oversight in the parameter file on my side. In the end, I most likely want to create exactly as many tracers as are present in my file. 
This comment is still outdated though, because I moved the `Number of particles` parameter into the individual plugins. Therefore, there is no such parameter, which could be ignored by this plugin. In the end I would prefer to keep it the current way, but update the documentation and make it more clear. Are you ok with that?